### PR TITLE
401 from upstream server should be 502

### PR DESCRIPTION
**Pull request by kreek**

### CODE REVIEW

File: lib/common/client/concerns/service_errors.rb [hunk 1]
```diff
@@ -14,7 +14,9 @@ def handle_service_error(error)
     def handle_error(error)
       case error.status
       when 401
-        raise Common::Exceptions::Unauthorized, error_details('The upstream server responded 401 Unauthorized')
+        raise Common::Exceptions::BadGateway, error_details(
+          "Received an unauthorized (401) response from the upstream server, check this error's source for event id"
```
Comment (saneshark) [inline lib/common/client/concerns/service_errors.rb]: NITPICK:

Possibly consider making the message more consistent with the message for 403 below:

"The upstream server responded with 401 Unauthorized"

I think checking the errors source for event id etc might go without saying?

File: lib/common/client/concerns/service_errors.rb [hunk 2]
```diff
@@ -33,7 +33,7 @@ def handle_error(error)
     end
 
     def log_original_error(error)
-      level, message = if (400...500).cover? error.status
+      level, message = if (401...500).cover? error.status
```
Comment (patrickvinograd) [inline lib/common/client/concerns/service_errors.rb]: Why did you change this line?

Comment (kreek) [reply lib/common/client/concerns/service_errors.rb]: ack, that should be 402...500, but the intent was to move 401 out of the expected response codes to use the unexpected method.

Comment (patrickvinograd) [reply lib/common/client/concerns/service_errors.rb]: I personally find it maddening that ruby has a `..` and `...` operator.
